WPMedia/fusion-news-theme-blocks
# OIDC for login block and social sign in

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

Update the Identity block for login and social sign in. This will add the capability to use PKCE and OIDC to a clients website. This allows them to use an ArcXP org/site as an auth provider.

## Jira Ticket

- [https://arcpublishing.atlassian.net/browse/ASUB-7845](https://arcpublishing.atlassian.net/browse/ASUB-7845)

## Acceptance Criteria

- Should redirect back to the 3rd party site when user has been logged in. The redirect should include a code and state query param.
- Should not redirect back to 3rd party site unless PKCE query params exists.
- Should redirect back to 3rd party site without a user needing to submit the login form if user is already logged in.

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. You will need to run ref-ui from the `arc-subs-ui` repo. The branch you will need to checkout is below.
- https://github.com/WPMedia/arc-subs-ui/tree/ASUB-7859/sdk-identity-oidc
2. Check out `oidc-login` branch for `arc-themes-blocks`
3. Check out `oidc-login` branch for `arc-themes-feature-pack`
4. Replace this line code https://github.com/WPMedia/arc-subs-ui/blob/ac704fc3c0e824d7fc620c8fbe9b15f4132f271f/packages/sdk-ui-reference/src/components/OIDCPKCE/buttons/LogIn.js#L21 with below so it point to your localhost PageBuilder instance.
```
window.location.href = `http://localhost/pf/login/?_website=the-daily-intelligencer&${queryString}`;
```
5. Setup a login page and add both the `Identity Log In - Arc Block` and `Identity Social Sign On - Arc Block` blocks.
6. Make sure you check the `Login with OIDC PKCE` option for both blocks.
<img width="352" alt="Screenshot 2023-11-17 at 9 55 52 PM" src="https://github.com/WPMedia/arc-themes-blocks/assets/5377648/59d81022-0f8c-4a9a-bc92-e07b01948fc4">

7. Go to https://localhost:3000/#/oidc-login/
8. Set your API Origins to
- Identity: https://subssinglesite-the-daily-intelligencer-sandbox.api.cdn.arcpublishing.com
- Org: subssinglesite
- Site: the-daily-intelligencer
9. Click the login button. This should redirect you to the local PageBuilder login page.
<img width="1229" alt="Screenshot 2023-11-17 at 10 01 06 PM" src="https://github.com/WPMedia/arc-themes-blocks/assets/5377648/d9ff7ba5-9a81-41cf-9604-791df0357518">

10. If you don't have a login user for subssingle-site you will need to create one. You can create one at https://localhost:3000/#/register.
11. Log in and this should redirect you back to https://localhost:3000/#/oidc-login/ and display your auth tokens.
12. You should be able to get entitlements, extend your session, get entitlements, and get the user profile.
13. This should also work for social sign in.
14. Note: user should be redirected immediately if the are still logged into the ArcXP site. You will need to clear the localstorage or session storage to start over again.

## Effect Of Changes

### Before

_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After

_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
